util/src/main/java/org/hyperledger/besu/util/io/E2StoreReader.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright contributors to Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.util.io;
+
+import org.hyperledger.besu.util.e2.E2BeaconState;
+import org.hyperledger.besu.util.e2.E2BlockIndex;
+import org.hyperledger.besu.util.e2.E2ExecutionBlockBody;
+import org.hyperledger.besu.util.e2.E2ExecutionBlockHeader;
+import org.hyperledger.besu.util.e2.E2ExecutionBlockReceipts;
+import org.hyperledger.besu.util.e2.E2SignedBeaconBlock;
+import org.hyperledger.besu.util.e2.E2SlotIndex;
+import org.hyperledger.besu.util.e2.E2StoreReaderListener;
+import org.hyperledger.besu.util.e2.E2Type;
+import org.hyperledger.besu.util.snappy.SnappyFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.common.primitives.Bytes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xerial.snappy.SnappyFramedInputStream;
+
+/** Reads E2 store file such a .era and .era1 */
+public class E2StoreReader {
```

Comment:
Think it would be nicer to have this in the e2 package as it's not general io it's specific to era